packages/components/src/components/dialog/dialog.jsx

```diff
@@ -6,126 +6,122 @@ import { CSSTransition } from 'react-transition-group';
 import Button from '../button/button.jsx';
 import Icon from '../icon/icon.jsx';
 
-class Dialog extends React.Component {
-    componentDidMount() {
-        if (this.props.is_visible && this.props.disableApp) {
-            this.props.disableApp();
+const Dialog = ({
+    disableApp,
+    enableApp,
+    is_closed_on_cancel,
+    is_closed_on_confirm,
+    is_visible,
+    onCancel,
+    onConfirm,
+    ...other_props
+}) => {
+    React.useEffect(() => {
+        if (is_visible && disableApp) {
+            disableApp();
         }
-    }
-
-    componentDidUpdate() {
-        if (this.props.is_visible && this.props.disableApp) {
-            this.props.disableApp();
-        }
-    }
+    });
```

Comment:
useEffect without dependency array may result in infinite execution. can you pls reconsider.

Reply:
Hmm, isn't it the expected behavior as `componentDidUpdate` is supposed to be executed constantly?

Reply:
Yes, so it needs to render every time the dependencies are changing

`
React.useEffect(() => {
        if (is_visible && disableApp) {
            disableApp();
        }
}, [is_visible, disableApp]);
`

Reply:
Alright, great suggestion. I added dependency array with the properties.